lib/rules/no-extra-boolean-cast.js

```diff
@@ -79,20 +132,16 @@ module.exports = {
                     parent = ancestors.pop(),
                     grandparent = ancestors.pop();
 
+
                 // Exit early if it's guaranteed not to match
                 if (node.operator !== "!" ||
-                        parent.type !== "UnaryExpression" ||
-                        parent.operator !== "!") {
+                          parent.type !== "UnaryExpression" ||
+                          parent.operator !== "!") {
                     return;
                 }
 
-                if (isInBooleanContext(parent, grandparent) ||
 
-                    // Boolean(<bool>) and new Boolean(<bool>)
-                    ((grandparent.type === "CallExpression" || grandparent.type === "NewExpression") &&
-                        grandparent.callee.type === "Identifier" &&
-                        grandparent.callee.name === "Boolean")
-                ) {
+                if (isInFlaggedContext(parent, grandparent)) {
```

Comment:
I think that `!!` fixer should now check precedence, similar to how `Boolean()` fixer already does.

For instance:

 ```js
if (a && !!(b ? c : d)) {
  // ...
}
```

would be probably fixed to:

 ```js
if (a && b ? c : d) {
  // ...
}
```

which changes semantics.

Actually, this is already a bug in the actual version, e.g with `!!(a = b) ? c : d`